### ISSUE TITLE
Unable to verify `ipfs daemon` is working on OS X

### ISSUE THREAD

**Issue by alco** [ipfs/go-ipfs#1737]
I have a problem where I can start the daemon but cannot access `webui` nor see any node connections.

My setup:

```
$ ipfs version
ipfs version 0.3.8-dev

$ ipfs id
{
	"ID": "QmRHuP8ai1wQip36VDckShFMbqvPi6oegf19uRWLVqUBpo",
	"PublicKey": "CAASpgIwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDYOFFUhk2qtcLrb0ijHY8Wksee6atu87Lkl1OJr+jwk3YDcFp1+st2ydeNIMm1NDeKVgkjurBS7AuoKZt9jfOAEgjryAoBViu8sg0CVT0Xc8iybYYT0rtaWxKl2w6d1D+YbNwosth9XcUrIVr2PltE18TZT3ggrXOqoCii6fitRfNhv6O5OltwH7GOEwN+HMQASiEGJDxI9+P68M9NaOSU5iEkuFwrvBbBY7tM0rNVHEHohtfF+9OjV7MQj003EX0Sn5E1cuzX9G7+9oe6+XZTVw+tEVPAj3ut0nzuc2d8UUe7ZI4+i8D6VJPN0cAW6VWIF1GjoDbez4RYidXBNUJ1AgMBAAE=",
	"Addresses": [
		"/ip4/127.0.0.1/tcp/4001/ipfs/QmRHuP8ai1wQip36VDckShFMbqvPi6oegf19uRWLVqUBpo",
		"/ip4/192.168.0.103/tcp/4001/ipfs/QmRHuP8ai1wQip36VDckShFMbqvPi6oegf19uRWLVqUBpo",
		"/ip6/::1/tcp/4001/ipfs/QmRHuP8ai1wQip36VDckShFMbqvPi6oegf19uRWLVqUBpo",
		"/ip4/31.43.110.205/tcp/64651/ipfs/QmRHuP8ai1wQip36VDckShFMbqvPi6oegf19uRWLVqUBpo"
	],
	"AgentVersion": "go-ipfs/0.3.8-dev",
	"ProtocolVersion": "ipfs/0.1.0"
}

$ ipfs config Addresses
{
  "API": "/ip4/127.0.0.1/tcp/5001",
  "Gateway": "/ip4/0.0.0.0/tcp/8080",
  "Swarm": [
    "/ip4/0.0.0.0/tcp/4001",
    "/ip6/::/tcp/4001"
  ]
}

$ ipfs daemon
Initializing daemon...
Swarm listening on /ip4/127.0.0.1/tcp/4001
Swarm listening on /ip4/192.168.0.103/tcp/4001
Swarm listening on /ip4/31.43.110.205/tcp/59258
Swarm listening on /ip6/::1/tcp/4001
API server listening on /ip4/127.0.0.1/tcp/5001
Gateway (readonly) server listening on /ip4/0.0.0.0/tcp/8080
Daemon is ready


$ curl -i -L http://localhost:5001/webui
HTTP/1.1 301 Moved Permanently
Location: /webui/
Date: Mon, 21 Sep 2015 22:14:11 GMT
Content-Length: 42
Content-Type: text/html; charset=utf-8

HTTP/1.1 302 Found
Location: /ipfs/QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6
Date: Mon, 21 Sep 2015 22:14:11 GMT
Content-Length: 75
Content-Type: text/html; charset=utf-8

HTTP/1.1 302 Found
Cache-Control: public, max-age=29030400
Etag: QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6
Location: /ipfs/QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6/
Suborigin: QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6
X-Ipfs-Path: /ipfs/QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6
Date: Mon, 21 Sep 2015 22:14:11 GMT
Content-Length: 76
Content-Type: text/html; charset=utf-8

HTTP/1.1 500 Internal Server Error
Cache-Control: public, max-age=29030400
Etag: QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6
Suborigin: QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6
X-Ipfs-Path: /ipfs/QmS2HL9v5YeKgQkkWMvs1EMnFtUowTEdFfSSeMT4pos1e6/
Date: Mon, 21 Sep 2015 22:15:11 GMT
Content-Length: 43
Content-Type: text/plain; charset=utf-8

internalWebError: context deadline exceeded⏎
~ 60.015 sec
```

It appears to be caught in a redirect loop and times out with a 500 after one minute,

I also don't see any output from `ipfs swarm peers`. Is there a public node to which I could connect manually to check that I can at least establish a connection to another node from my machine?

**Comment by jbenet** [comment]
I also don't see any output from `ipfs swarm peers` this is the problem. Sounds like you're not reaching the bootstrap nodes. make sure the bootstrap list is correct-- you should see:

```sh
> ipfs bootstrap
/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ
/ip4/104.236.151.122/tcp/4001/ipfs/QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx
/ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z
/ip4/104.236.179.241/tcp/4001/ipfs/QmSoLpPVmHKQ4XTPdz8tjDFgdeRFkpV8JgYq8JVJ69RrZm
/ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64
/ip4/128.199.219.111/tcp/4001/ipfs/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu
/ip4/162.243.248.213/tcp/4001/ipfs/QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm
/ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd
/ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3
```

**Comment by alco** [comment]
Here's my output, it looks identical to yours. I have also tried disabling the system firewall with no luck.

```
$ ipfs bootstrap
/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ
/ip4/104.236.151.122/tcp/4001/ipfs/QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx
/ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z
/ip4/104.236.179.241/tcp/4001/ipfs/QmSoLpPVmHKQ4XTPdz8tjDFgdeRFkpV8JgYq8JVJ69RrZm
/ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64
/ip4/128.199.219.111/tcp/4001/ipfs/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu
/ip4/162.243.248.213/tcp/4001/ipfs/QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm
/ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd
/ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3
```

**Comment by jbenet** [comment]
@alco anything odd about your network? can you try this is another machine, like a VPS somewhere?

**Comment by alco** [comment]
I just tried again and got `internalWebError: context deadline exceeded` immediately. Then I removed my `$GOPATH` and fetched `go-ipfs` again using `go get` and now webui works.

**Comment by alco** [comment]
And I now also see 94 peers.

**Comment by alco** [comment]
Does it mean any IPFS user anywhere in the world can now cat my cat pictures?

**Comment by whyrusleeping** [comment]
@alco well, we would be able to if we knew the hashes.

**Comment by jbenet** [comment]
Sounds like this is almost fixed. remaining:

- [ ] @alco gives us the cat hashes.

**Comment by alco** [comment]
I have edited this file so it should be a unique copy on the network. If you can load it, then we have a contact: http://gateway.ipfs.io/ipfs/QmTJHuzG3mjgmvcfvTU4ykWXwD4QjA5aCk6QsU4BPaD8Hh/cat2.gif

**Comment by chriscool** [comment]
I can load it :-)

**Comment by Mithgol** [comment]
However, the gate seems to time out (`Path Resolve error: context deadline exceeded`) on most of the PNG files in the https://ipfs.io/ipfs/QmTJHuzG3mjgmvcfvTU4ykWXwD4QjA5aCk6QsU4BPaD8Hh directory. At least currently.

**Comment by alco** [comment]
@Mithgol I didn't think that anyone would try listing the directory and brought my node down half an hour ago. I have now brought it back up.

**Comment by jbenet** [comment]
:+1: for viewable cats. 

---

![cat.gif](https://ipfs.io/ipfs/QmTJHuzG3mjgmvcfvTU4ykWXwD4QjA5aCk6QsU4BPaD8Hh/cat.gif)

**Comment by RichardLitt** [comment]
> I just tried again and got internalWebError: context deadline exceeded immediately (in Chrome). Then I removed my $GOPATH and fetched go-ipfs again using go get and now webui works.

Do we expect this to be a common enough problem to add somewhere?

**Comment by jbenet** [comment]
well, i don't even know what happened-- pretty odd. lets maybe see if anybody else runs into it before putting it anywhere.

**Comment by anatol** [comment]
I use Linux Arch and see similar symptoms - daemon is happy, I see bootstrap nodes but no output from `ipfs swarm peers`. Any attempt to make `ipfs cat XXX` for a remote content hangs (as there are no connections to peers).

I use these build instructions for the binary https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=go-ipfs
Is there anything wrong with these instructions?

**Comment by anatol** [comment]
False alarm guys. I just messed up with my ipfs config. After fixing it `ipfs` works fine and I can add/get files without issues.

**Comment by whyrusleeping** [comment]
what was wrong with your config? if you dont mind sharing for future people who have similar symptoms.

**Comment by anatol** [comment]
I installed `ipfs` at my Linux system and found that port 4001 is already taken and thus daemon did not start saying `bind: port is is use` error. I edited `~/.ipfs/config` and replaced the port to other unused port (4007) but I also accidentally replaced port at bootstrap peers. Daemon started, no errors was reported to logs but I have not seen any peers. After generating a clean config file and comparing to mine I quickly found the problem.

Having some kind of diagnostic tool or a better logging for this case might be useful.
